### ISSUE TITLE
Devices in /dev/<zpoolname>/<zvolname> sometimes are not available immediately after created zvol

### ISSUE THREAD

**Issue by arturpzol** [zfsonlinux/zfs#8669]
### System information

Type | Version/Name
 --- | --- 
Distribution Name	| Debian
Distribution Version	|  Jessie
Linux Kernel	| 4.19.36
Architecture	| x86_64
ZFS Version	| 0.8.0-rc4
SPL Version	| 0.8.0-rc4

### Describe the problem you're observing
Sometimes I need to wait a moment for device after created new zvol.
Please see below commands dump:

```
zfs create -V 4G Pool-0/zvol1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.2 ; ls -al /dev/Pool-0/zvol1
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
```
and next execution:

```
zfs destroy Pool-0/zvol1
zfs create -V 4G Pool-0/zvol1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.2 ; ls -al /dev/Pool-0/zvol1
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
lrwxrwxrwx 1 root root 7 Apr 25 12:13 /dev/Pool-0/zvol1 -> ../zd16
```
and next:

```
zfs destroy Pool-0/zvol1
zfs create -V 4G Pool-0/zvol1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.2 ; ls -al /dev/Pool-0/zvol1
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
```
`udevadm settle` does not help:
```

zfs create -V 4G Pool-0/zvol1 ; ls -al /dev/Pool-0/zvol1 ; sleep 0.1 ; udevadm settle; ls -al /dev/Pool-0/zvol1 ; sleep 0.2 ; ls -al /dev/Pool-0/zvol1
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
ls: cannot access /dev/Pool-0/zvol1: No such file or directory
```

I have tried with libudev and without it but effect is the same:

```
checking libudev.h usability... yes
checking libudev.h presence... yes
checking for libudev.h... yes
checking for library containing udev_device_get_is_initialized... -ludev
```

Issue can be solved (in scripting) by adding longer sleep but of course it will be workaround. Should `udev_device_get_is_initialized` waits also for dev creating? 

In ZFS 0.6 issue can be also reproduced but `udevadm seettle` helps at least:

```
zfs create -V 4G Pool-1/zvol1 ; ls -al /dev/Pool-1/zvol1 ; udevadm settle ; ls -al /dev/Pool-1/zvol1 
ls: cannot access /dev/Pool-1/zvol1: No such file or directory
lrwxrwxrwx 1 root root 7 Apr 25 12:37 /dev/Pool-1/zvol1 -> ../zd16
```

**Comment by behlendorf** [comment]
There's some work being done in PR #8615 which should help speed this up.

**Comment by arturpzol** [comment]
@behlendorf seems that fix in #8615 solved my issue. Thank you for the information about the fix.
